sdks/java/nexmark/src/main/java/org/apache/beam/sdk/nexmark/NexmarkLauncher.java

```diff
@@ -1023,8 +1037,15 @@ private void sinkResultsToBigQuery(
             break;
           case SUBSCRIBE_ONLY:
           case COMBINED:
-            // Read events from pubsub.
-            source = sourceEventsFromPubsub(p, now);
+            {
+              // Read events from Kafka or Pubsub.
+              if (configuration.sourceType == NexmarkUtils.SourceType.KAFKA) {
+                source =
+                    sourceEventsFromKafka(p, configuration.pubSubMode == COMBINED ? now : null);
```

Comment:
Please add a comment about why we pass now based on `pubsubMode`

Reply:
Sure, good point, added.